src/history.rs

```diff
@@ -0,0 +1,46 @@
+use smart_debug::SmartDebug;
+use std::path::PathBuf;
+
+#[derive(SmartDebug, Clone, PartialEq)]
+pub struct History {
+    history: Vec<PathBuf>,
+    index: usize,
+}
+
+impl History {
+    pub fn new(path_buf: PathBuf) -> Self {
+        History {
+            history: vec![path_buf],
+            index: 0,
+        }
+    }
+    pub fn truncate(&mut self) {
+        if self.index + 1 < self.history.len() {
+            self.history.truncate(self.index + 1);
+        }
+    }
+    pub fn append(&mut self, file_path: PathBuf) {
+        self.history.push(file_path);
+        self.index += 1;
+    }
+    pub fn get_path(&self) -> &PathBuf {
+        self.history
+            .get(self.index)
+            .expect("History should be bound checked for all possible indexes.")
+    }
+
+    pub fn next(&mut self) -> Option<&PathBuf> {
```

Comment:
```suggestion
    #[allow(clippy::should_implement_trait)]
    pub fn next(&mut self) -> Option<&PathBuf> {
```

I think the name is fine here